moshi/src/main/java/com/squareup/moshi/StandardJsonAdapters.java

```diff
@@ -215,21 +214,25 @@ static int rangeCheckNextInt(JsonReader reader, String typeMessage, int min, int
 
   static final class EnumJsonAdapter<T extends Enum<T>> extends JsonAdapter<T> {
     private final Class<T> enumType;
-    private final Map<String, T> nameConstantMap = new LinkedHashMap<>();
+    private final Map<String, T> nameConstantMap;
+    private final String[] nameStrings;
 
     public EnumJsonAdapter(Class<T> enumType) {
       this.enumType = enumType;
       try {
         T[] constants = enumType.getEnumConstants();
-        for (T constant : constants) {
+        nameConstantMap = new LinkedHashMap<>();
```

Comment:
It’s surprisingly hard work. http://google.github.io/guava/releases/snapshot/api/docs/src-html/com/google/common/collect/Maps.html#line.186